docx-core/examples/footnotes.rs

```diff
@@ -0,0 +1,20 @@
+use docx_rs::*;
+
+pub fn main() -> Result<(), DocxError> {
+    let path = std::path::Path::new("./footnotes.docx");
+    let file = std::fs::File::create(path).unwrap();
+    Docx::new()
+        .add_paragraph(
+            Paragraph::new()
+                .add_run(Run::new().add_text("Hello"))
+                .add_run(
+                    Run::new().add_footnote_reference(
+                        Footnote::new(1)
+                            .add_content(Paragraph::new().add_run(Run::new().add_text("World"))),
+                    ),
+                ),
```

Comment:
@git-noise 
Thanks for your great work!!!!!!

[question]
In `docx-rs`, I'm using an interface extensively that automatically assigns IDs. Can this feature be implemented to automatically assign IDs as well, in the following interface?"

``` rust
Run::new().add_footnote(
  Footnote::new().add_content(Paragraph::new().add_run(Run::new().add_text("World"))),
)
```

ref:  https://github.com/bokuweb/docx-rs/blob/main/docx-core/src/documents/hyperlink_id.rs

Reply:
Hello,

I did the modification, for now only on `Footnotes` themselves not `FootnoteReference`.

The reasoning is that in the current code, while one can technically instantiate a `FootnoteReference` by directly calling `FootnoteReference::new()`, the intended usage - also showcased in the example you mentioned above - is to use `add_footnote_reference()` on a `Footnote`.
Behind the scene, a `FootnoteReference` is then created from `Footnotes` by calling `into()`.

If we want to also add automatic IDs for FootnoteReference themselves, then we may hit a synchronization issue, as we may need to keep the FootnoteReference ID and Footnote ID in sync.

The current setup probably strikes a balance between usability and simplicity, but let me know your thoughts.

Reply:
@git-noise 

Thank you. In cases where the user does not need to be aware of the id, I want the id to be hidden and added automatically. Other functions in this library are designed this way. Therefore, if possible, I would like the id to be auto-generated.

Reply:
I think it currently is the case. With the latest modification, a user would call something like the following:
```rust
Run::new().add_footnote_reference(
  Footnote::new().add_content(
    Paragraph::new().add_run(Run::new().add_text("World"))
  )
)
```
In this case there is no need to give an ID, and behind the scene, the `Footnote` and `FootnoteReference` are created with the same ID.

Reply:
@git-noise Sorry! thanks for your work :)